src/liveness.zig

```diff
@@ -144,6 +144,91 @@ fn analyzeInst(
             // instruction, and the deaths flag for the CondBr instruction will indicate whether the
             // condition's lifetime ends immediately before entering any branch.
         },
+        .switchbr => {
+            const inst = base.castTag(.switchbr).?;
+
+            const Table = std.AutoHashMap(*ir.Inst, void);
+            const case_tables = try table.allocator.alloc(Table, inst.cases.len + 1); // +1 for else
+            defer table.allocator.free(case_tables);
+
+            std.mem.set(Table, case_tables, Table.init(table.allocator));
+            defer for (case_tables) |*ct| ct.deinit();
+
+            for (inst.cases) |case, i| {
+                try analyzeWithTable(arena, table, &case_tables[i], case.body);
+
+                // Reset the table back to its state from before the case.
+                var it = case_tables[i].iterator();
+                while (it.next()) |entry| {
+                    table.removeAssertDiscard(entry.key);
+                }
+            }
+            { // else
+                try analyzeWithTable(arena, table, &case_tables[case_tables.len - 1], inst.else_body);
+
+                // Reset the table back to its state from before the case.
+                var it = case_tables[case_tables.len - 1].iterator();
+                while (it.next()) |entry| {
+                    table.removeAssertDiscard(entry.key);
+                }
+            }
+
+            const List = std.ArrayList(*ir.Inst);
+            const case_deaths = try table.allocator.alloc(List, case_tables.len); // +1 for else
+            defer table.allocator.free(case_deaths);
+
+            std.mem.set(List, case_deaths, List.init(table.allocator));
+            defer for (case_deaths) |*cd| cd.deinit();
+
+            var total_deaths: u32 = 0;
+            for (case_tables) |*ct, i| {
+                total_deaths += ct.count();
+                var it = ct.iterator();
+                while (it.next()) |entry| {
+                    const case_death = entry.key;
+                    for (case_tables) |*ct_inner, j| {
+                        if (i == j) continue;
+                        if (!ct_inner.contains(case_death)) {
```

Comment:
Hmm this looks `O(N ^ 2)`. This might be OK but I'd like to think about it some more, see how it performs with large switch expressions, and maybe try to come up with another idea for how to do this if it turns out to be a cause of compilation slowdowns.

BTW were you able to figure out how to get liveness analysis debug output so you could see the results of this code in action?

Reply:
>Hmm this looks `O(N ^ 2)`. 

I've been focusing on getting correct behavior, hopefully this can be optimized much better.

> BTW were you able to figure out how to get liveness analysis debug output so you could see the results of this code in action?

Yes, seems mostly correct after that last commit.